src/Centreon/Application/Controller/Monitoring/SubmitResultController.php

```diff
@@ -0,0 +1,273 @@
+            '/../../../../../config/json_validator/latest/Centreon/SubmitResult/SubmitResult.json';
+        $validator->validate(
+            $bodyContent,
+            (object) ['$ref' => $file],
+            Constraint::CHECK_MODE_VALIDATE_SCHEMA
+        );
+
+        if (!$validator->isValid()) {
+            $message = '';
+            foreach ($validator->getErrors() as $error) {
+                $message .= sprintf("[%s] %s\n", $error['property'], $error['message']);
+            }
+            throw new SubmitResultException($message);
+        }
+
+        /**
+         * At this point we made sure that the mapping will work since we validate
+         * the JSON sent with the JSON validator.
+         */
+        $status = ResourceStatus::STATUS_MAPPING[strtoupper($results['status'])];
+
+        $result = (new SubmitResult($hostId, $status))
+            ->setOutput($results['output'])
+            ->setPerformanceData($results['performance_data']);
+
+        $this->submitResultService
+            ->filterByContact($contact)
+            ->submitHostResult($result);
+
+        return $this->view();
+    }
+
+    /**
+     * Entry point to submit result to a service.
+     *
+     * @param Request $request
+     * @param int $hostId ID of service parent (host)
+     * @param int $serviceId ID of the service
+     * @return View
+     * @throws \Exception
+     */
+    public function submitResultService(
+        Request $request,
+        int $hostId,
+        int $serviceId
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+
+        /**
+         * @var Contact $contact
+         */
+        $contact = $this->getUser();
+        if (!$contact->isAdmin() && !$contact->hasRole(Contact::ROLE_SERVICE_SUBMIT_RESULT)) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        $results = json_decode((string) $request->getContent(), true);
+        if (!is_array($results)) {
+            throw new \InvalidArgumentException(_('Error when decoding sent data'));
+        }
+
+        /*
+        * Validate the content of the POST request against the JSON schema validator
+        */
+        $validator = new Validator();
+        $bodyContent = json_decode((string) $request->getContent());
+        $file = 'file://' . __DIR__ .
+            '/../../../../../config/json_validator/latest/Centreon/SubmitResult/SubmitResult.json';
+        $validator->validate(
+            $bodyContent,
+            (object) ['$ref' => $file],
+            Constraint::CHECK_MODE_VALIDATE_SCHEMA
+        );
+
+        if (!$validator->isValid()) {
+            $message = '';
+            foreach ($validator->getErrors() as $error) {
+                $message .= sprintf("[%s] %s\n", $error['property'], $error['message']);
+            }
+            throw new SubmitResultException($message);
+        }
+
+        $status = ResourceStatus::STATUS_MAPPING[strtoupper($results['status'])];
```

Comment:
all this section seems exactly the same within `submitResultHost` method
you should extract this in a separated private method